src/CallbackInvoker.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use DI\Bridge\Silex\Application;
+use Invoker\Invoker;
+use Interop\Container\ContainerInterface;
+use Invoker\ParameterResolver\ResolverChain;
+use Invoker\ParameterResolver\TypeHintResolver;
+use Invoker\ParameterResolver\AssociativeArrayResolver;
+use Invoker\ParameterResolver\Container\TypeHintContainerResolver;
+use Invoker\ParameterResolver\NumericArrayResolver;
+
+/**
+ * A subclass of Invoker that always tries to first resolve through provided parameter names, then
+ * type hints, then through the DI container and finally allows a fallback to a default parameter order
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class CallbackInvoker extends Invoker
+{
+    /**
+     * @param ContainerInterface $container the container for injection
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        parent::__construct(new ResolverChain([
+            new AssociativeArrayResolver,
+            new TypeHintResolver,
+            new TypeHintContainerResolver($container),
+            new NumericArrayResolver,
+        ]));
+    }
```

Comment:
I'm not convinced by the usefulness of this class. I think you referred to that when you said:

> It extends Invoker for type safety so MiddlewareListener etc. can depend on this specific Invoker with this specific ResolverChain.

But I don't see why a MiddlewareListener should explicitly require this implementation. To me it would be better than middleware listeners or parameter converters are implementations that rely on the interface of "Invoker", and then the application passes an instance of `Invoker` configured as we want.

I.e. I don't see the point to couple things more than necessary.